refactor(linter): remove `#[must_use]` from `LintService::with_*` methods

overlookmotel opened the pull request:
Pure refactor. Remove `#[must_use]` from `LintService::with_paths` and `LintService::with_file_system`. I assume these are left over from a past time when these methods took and returned `self`.

Also, make these same methods on `Runtime` return `&mut Self` not `&Self`. We don't use the return value of these methods, but presumably they're meant to support chaining.

camc314 commented:
### Merge activity

* **Jul 28, 9:05 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(linter): remove `#[must_use]` from `LintService::with_*` methods (#12560)

Pure refactor. Remove `#[must_use]` from `LintService::with_paths` and `LintService::with_file_system`. I assume these are left over from a past time when these methods took and returned `self`.

Also, make these same methods on `Runtime` return `&mut Self` not `&Self`. We don't use the return value of these methods, but presumably they're meant to support chaining.
